packages/calypso-e2e/src/lib/blocks/block-flows/markdown.ts

```diff
@@ -0,0 +1,66 @@
+import { BlockFlow, EditorContext, PublishedPostContext } from '.';
+
+interface ConfigurationData {
+	text: string;
+}
+interface ValidationData {
+	expectedText: string;
+	// Has to be one of the aria roles supported by the locator.getByRole call.
+	expectedRole: 'heading' | 'paragraph';
+}
+
+/**
+ * Represents the flow of using the Markdown block.
+ */
+export class MarkdownFlow implements BlockFlow {
+	private configurationData: ConfigurationData;
+	private validationData: ValidationData;
+
+	/**
+	 * Constructs an instance of this block flow with data to be used when
+	 * configuring and validating the block.
+	 *
+	 * @param {ConfigurationData} configurationData Configuration data for the block.
+	 */
+	constructor( configurationData: ConfigurationData, validationData: ValidationData ) {
+		this.configurationData = configurationData;
+		this.validationData = validationData;
+	}
+
+	blockSidebarName = 'Markdown';
+	blockEditorSelector = 'div[aria-label="Block: Markdown"]';
+
+	/**
+	 * Configure the block in the editor with the configuration data from the constructor
+	 *
+	 * @param {EditorContext} context The current context for the editor at the point of
+	 * test execution.
+	 */
+	async configure( context: EditorContext ): Promise< void > {
+		const editorCanvas = await context.editorPage.getEditorCanvas();
+		const block = editorCanvas.getByRole( 'document', {
+			name: `Block: ${ this.blockSidebarName }`,
+		} );
```

Comment:
**Nit (non-blocking)**: As an FYI -- there's now a locator that points to the block parent element already in the `EditorContext` object, so you should be able to just use that. 🙂

Reply:
Gotcha, thanks for the heads up - I replaced both the Markdown and Related Posts flows with the Locator in the context.